cmd/agent.go

```diff
@@ -16,7 +16,7 @@ var agentCmd = &cobra.Command{
 	Short: "start the preflight agent",
 	Long: `The agent will periodically gather data for the configured data
 	gatherers and send it to a remote backend for evaluation`,
-	Run: agent.Run,
+	RunE: agent.Run,
```

Comment:
I'll break this PR up in to some smaller chunks.
Changing these to `RunE` allows me to remove all the use of `log.Fatal` in the code.

Reply:
See https://github.com/jetstack/jetstack-secure/pull/599/files#r1822046405